server/unittools.cpp

```diff
@@ -1409,27 +1413,43 @@ void bounce_unit(struct unit *punit, bool verbose, bounce_reason reason,
    * Try to bounce transported units. */
   if (0 < get_transporter_occupancy(punit)) {
     const auto pcargo_units = unit_transport_cargo(punit);
-    unit_list_iterate(pcargo_units, pcargo) { bounce_unit(pcargo, verbose); }
+    unit_list_iterate(pcargo_units, pcargo)
+    {
+      bounce_unit(pcargo, max_distance, on_success, on_failure);
```

Comment:
Here we are in a tricky situation, because the script running in the move handler might have killed everything on the tile: "unloading a unit may trigger a nuclear explosion" sounds fitting for the Chaos ruleset. I think we need `unit_list_iterate_safe` and check that the cargo still exists afterwards.

Reply:
I suspect that `pcargo_units` will be freed if `pcargo` gets killed in an event handler (and hopefully the loaded units will have been deleted before). Also, you need to check if `pcargo` still exists before calling `on_failure` or `wipe_unit` with it.

Reply:
Sorry, I thought the transported unit list being deleted would also create issues, but on reading  `unit_list_iterate_safe` I see that it is copied before iterating. So we're safe from that.